### ISSUE TITLE
Separate mousewheel-scroll setting & add secondary zoom method

### ISSUE THREAD

**Issue by themoenen** [epam/miew#557]
**Is your feature request related to a problem? Please describe.**
Zooming in and out can only be done with the mousewheeel. This can be disrupting to the scroll experience.

**Describe the solution you'd like**
1. Allow user to disable mousewheel zoom without also disabling pinch-to-zoom
2. Provide a secondary method for zooming that doesn't involve the mousewheel, e.g. ctrl-drag

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Comment by NataliaDSmirnova** [comment]
@themoenen , I was unable to replicate the reported bug while testing Miew in both TypeScript and TypeScript with React applications. It seems that the `any` type for options in the Miew constructor is enough to avoid any errors.

![image](https://github.com/user-attachments/assets/5515fdbf-b848-4e10-a6ff-26515709e740)

Could you please provide more details about your specific setup, including the version of Miew you're using, any relevant code snippets, and the exact steps to reproduce the issue? This information will help us narrow down the potential causes."

**Comment by themoenen** [comment]
I think this was supposed to be in response to #559. I will answer over there.

**Comment by paulsmirnov** [comment]
> *This was generated by AI during triage.*

## Agent Brief

**Category:** enhancement
**Summary:** Decouple mousewheel zoom from touch/drag zoom and add secondary mouse drag zooming

**Current behavior:**
The viewer uses a single boolean setting `zooming: true` in `SettingsObject` that controls both mouse wheel zooming and touch pinch-to-zoom in `ObjectControls`. Setting `zooming: false` completely disables all zoom capabilities. Furthermore, zooming cannot be performed via mouse drag (only mouse wheel is supported for mouse input).

**Desired behavior:**
1. The `zooming` setting should support granular channel configuration: `{ wheel: true, drag: true, touch: true }`.
2. Setting `zooming` as a boolean (`false` or `true`) must be normalized: `false` disables all channels (`{ wheel: false, drag: false, touch: false }`) and `true` enables all channels (`{ wheel: true, drag: true, touch: true }`). Individual channels can be toggled independently (e.g. `set('zooming.wheel', false)` disables only mousewheel zoom).
3. In `ObjectControls`:
   - Mouse wheel zooming is gated on `zooming.wheel`.
   - Multi-touch pinch zooming is gated on `zooming.touch`.
   - Secondary mouse drag zooming is gated on `zooming.drag`.
4. Secondary mouse drag zooming is triggered by:
   - `Ctrl + Left Click + Drag` (when `Alt` is not pressed).
   - Middle mouse button (`button === 1`) drag.
   - Vertical dragging movement scales the object (moving cursor up zooms in, moving down zooms out).

**Key interfaces:**
- `SettingsObject`:
  - `defaults.zooming`: `{ wheel: true, drag: true, touch: true }`.
  - Normalization logic in `Settings.set()` / constructor options for boolean values passed to `zooming`.
- `ObjectControls`:
  - `mousewheel()`: checks `settings.now.zooming.wheel`.
  - `touchmove()` for `STATE.SCALE`: checks `settings.now.zooming.touch`.
  - `mousedown()` / `mousemove()`: adds drag zoom state (triggered by `Ctrl + Left Click` without `Alt` or middle mouse button) that adjusts scale based on vertical delta and checks `settings.now.zooming.drag`.

**Acceptance criteria:**
- [ ] `defaults.zooming` is an object `{ wheel: true, drag: true, touch: true }`.
- [ ] Setting `zooming: false` disables wheel zoom, pinch zoom, and mouse drag zoom.
- [ ] Setting `zooming: true` enables wheel zoom, pinch zoom, and mouse drag zoom.
- [ ] Setting `zooming.wheel = false` disables mouse wheel zoom while pinch-to-zoom and mouse drag zoom continue to work.
- [ ] Setting `zooming.touch = false` disables pinch-to-zoom while mouse wheel and mouse drag zoom continue to work.
- [ ] `Ctrl + Left Click + Drag` (without `Alt`) zooms the scene in and out based on vertical mouse movement.
- [ ] Middle mouse button click and drag zooms the scene in and out based on vertical mouse movement.
- [ ] Unit tests cover `zooming` boolean normalization, settings diffing/merging, and `ObjectControls` channel handling.

**Out of scope:**
- Reconfiguring `rotating` or `translating` settings into similar channel objects.
- Adding UI buttons or on-screen zoom slider controls.
